### PR TITLE
Add filter plugin

### PR DESCRIPTION
**Pull request by okkez**

### CODE REVIEW

File: lib/fluent/plugin/query_params_extractor.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
+require 'uri'
+require 'cgi/util'
+require 'webrick'
```
Comment (kentaro) [inline lib/fluent/plugin/query_params_extractor.rb]: Has it been required yet?

Comment (okkez) [inline lib/fluent/plugin/query_params_extractor.rb]: Yes, we are using followings for now:

* `URI.parse`
* `WEBrick::HTTPUtils.escape`
* `CGI.unescape`

**Comment by kentaro** [comment]
How does fluentd detect whether to use `ExtractQueryParamsOutput` or `ExtractQueryParamsFilter`?

**Comment by kentaro** [comment]
Can you add some doc to README if needed?

**Comment by kentaro** [comment]
Ah, got it.

* When the fluentd `>= v0.12`, you can designate `ExtractQueryParamsFilter` in `<filter>` tag
* Otherwise, `ExtractQueryParamsOutput` in `<match>` tag is userd

Right?

File: lib/fluent/plugin/query_params_extractor.rb (hunk 1 above)
Comment (kentaro) [inline lib/fluent/plugin/query_params_extractor.rb]: Got it.

**Comment by kentaro** [comment]
Merged it tentatively. I'll soon write up a doc for the new version at another branch.

**Comment by okkez** [comment]
2015-09-07 18:26 GMT+09:00 Kentaro Kuribayashi <notifications@github.com>:

> Ah, got it.
>
>    - When the fluentd >= v0.12, you can designate ExtractQueryParamsFilter
>    in <filter> tag
>    - Otherwise, ExtractQueryParamsOutput in <match> tag is userd
>
> Right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kentaro/fluent-plugin-extract_query_params/pull/15#issuecomment-138249703>
> .
>

Yes, right.



-- 
okkez
okkez000@gmail.com

**Comment by kentaro** [comment]
Released the new version 0.0.11 https://rubygems.org/gems/fluent-plugin-extract_query_params/versions/0.0.11

**Pull request merged**